File: src/main/java/de/saschahlusiak/freebloks/database/FreebloksDBOpenHandler.java
package de.saschahlusiak.freebloks.database;
import android.content.Context;
import android.database.sqlite.SQLiteDatabase;
import android.database.sqlite.SQLiteOpenHelper;

public class FreebloksDBOpenHandler extends SQLiteOpenHelper {
	public static final String DATABASE_NAME = "freebloks.db";
	private static final int DATABASE_VERSION = 1;

	public FreebloksDBOpenHandler(Context context) {
		super(context, DATABASE_NAME, null, DATABASE_VERSION);
	}

	@Override
	public void onCreate(SQLiteDatabase db) {
		db.execSQL(HighscoreDB.CREATE_TABLE);
	}

	@Override
	public void onUpgrade(SQLiteDatabase db, int oldVersion, int newVersion) {
		try {

		} catch (Exception e) {
			db.execSQL(HighscoreDB.DROP_TABLE);
			onCreate(db);
		}
	}
}

File: src/main/java/de/saschahlusiak/freebloks/database/FreebloksDBOpenHandler.kt
package de.saschahlusiak.freebloks.database

import android.content.Context
import android.database.sqlite.SQLiteDatabase
import android.database.sqlite.SQLiteOpenHelper

class FreebloksDBOpenHandler(context: Context) : SQLiteOpenHelper(context, DATABASE_NAME, null, DATABASE_VERSION) {
    override fun onCreate(db: SQLiteDatabase) {
        db.execSQL(HighScoreDB.CREATE_TABLE)
    }

    override fun onUpgrade(db: SQLiteDatabase, oldVersion: Int, newVersion: Int) {
        try {
            // nothing yet
        } catch (e: Exception) {
            db.execSQL(HighScoreDB.DROP_TABLE)
            onCreate(db)
        }
    }

    companion object {
        private const val DATABASE_NAME = "freebloks.db"
        private const val DATABASE_VERSION = 1
    }
}